core/src/test/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraintsTest.java

```diff
@@ -0,0 +1,84 @@
+package fr.sncf.osrd.api.pathfinding.constraints;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.google.common.collect.Range;
+import fr.sncf.osrd.api.pathfinding.PathfindingRoutesEndpoint;
+import fr.sncf.osrd.api.pathfinding.response.NoPathFoundError;
+import fr.sncf.osrd.infra.api.Direction;
+import fr.sncf.osrd.stdcm.DummyRouteGraphBuilder;
+import fr.sncf.osrd.train.TestTrains;
+import fr.sncf.osrd.utils.graph.Pathfinding;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ElectrificationConstraintsTest {
+    static Stream<Arguments> testDeadSectionArgs() {
+        return Stream.of(
+                // How much catenary, with dead section, dead section direction, should block some ranges
+                Arguments.of(2, true, Direction.FORWARD, false),
+                Arguments.of(2, true, Direction.BACKWARD, false),
+                Arguments.of(2, false, Direction.FORWARD, false),
+                Arguments.of(1, true, Direction.FORWARD, false),
+                Arguments.of(1, true, Direction.BACKWARD, true),
+                Arguments.of(1, false, Direction.FORWARD, true),
+                Arguments.of(0, true, Direction.FORWARD, true),
+                Arguments.of(0, false, Direction.FORWARD, true)
```

Comment:
I know we've done the parametrized enum, but it's pretty hard to read. Could it be possible to pass the voltage map directly as a parameter?

If it could work out, the comment about making the logic explicit in the other parametrized function would also apply here.